cds/orgsvc/C#/UploadDownloadFilesImages/UploadDownloadFilesImages/SampleProgram.cs

```diff
@@ -0,0 +1,137 @@
+                                            </filter>
+                                            <link-entity name='publisher' from='publisherid' to='publisherid'>
+                                              <attribute name='uniquename' />
+                                              <attribute name='friendlyname' />
+                                              <attribute name='description' />
+                                              <attribute name='customizationprefix' />
+                                            </link-entity>
+                                          </entity>
+                                        </fetch>";
+
+            var publisherEntity =  service.RetrieveMultiple(new FetchExpression(publisherFetchXML))?.Entities.FirstOrDefault();
+            return ((AliasedValue)(publisherEntity?.Attributes["publisher1.customizationprefix"])).Value.ToString();
+        }
+
+        /// <summary>
+        /// Creates a new account record for a given crm service client
+        /// </summary>
+        /// <param name="service">A CrmServiceClient object</param>
+        /// <returns>A Guid for a newly created account record</returns>
+        private static Guid CreateNewAccountRecord(CrmServiceClient service)
+        {
+            var entity = new Entity("account", new Guid());
+            var accountName = "MyAccount" + DateTime.Now.ToFileTime();
+            entity.Attributes["name"] = accountName;
+            try
+            {
+                var accountId = service.Create(entity);
+                Console.WriteLine($"\nNew Account created: '{accountName} with Guid-{accountId}'");
+                return accountId;
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            } 
+        }
+
+
+        /// <summary>
+        /// Uplaods and downloads a file for a given CRM service client, a publisher customziation prefix and an account guid Id
+        /// </summary>
+        /// <param name="service"></param>
+        /// <param name="customizationPrefix"></param>
+        /// <param name="accountId"></param>
+        private static void UploadDownloadBlob(CrmServiceClient service, string customizationPrefix, Guid accountId)
+        {
+            var uploadDownload = new SampleBlobUploadDownload(service);
+            var entity = new Entity("account", accountId);
+
+            Dictionary<string, string> columnFiles = new Dictionary<string, string>();
+            columnFiles.Add($"{customizationPrefix}_file1", "Files\\25mb.pdf");
+            columnFiles.Add($"{customizationPrefix}_image1primary", "Images\\25mb.jpg");
+
+            foreach (var x in columnFiles)
+            {
+                var attributeName = x.Key;
+                
+                uploadDownload.Upload(entity, Path.GetFullPath(x.Value).Replace("\\bin\\Debug", ""), Path.GetFileName(x.Value), attributeName);             
```

Comment:
> ` uploadDownload.Upload(entity, Path.GetFullPath(x.Value).Replace("\\bin\\Debug", ""), Path.GetFileName(x.Value), attributeName);    `
 
 I think this could be simplified if the properties for the image in the solution were changed. Set **Copy to Output Directory** to **true**. Then you can simply refer to the file by name.